packages/compass-components/src/components/file-input.tsx

```diff
@@ -174,18 +204,40 @@ function FileInput({
           { [formItemErrorStyles]: error }
         )}
       >
-        <label
-          htmlFor={`${id}_file_input`}
+        <div
           className={cx(
-            { [labelHorizontalStyles]: variant === Variant.Horizontal },
+            {
+              [labelHorizontalStyles]: variant === Variant.Horizontal,
+            },
             css({
               textAlign: labelAlignment,
             })
           )}
         >
-          <span style={{ gridArea: 'label' }}>{label}</span>
+          <label
+            htmlFor={`${id}_file_input`}
+            className={cx({
+              [disabledLabelStyles]: disabled,
+            })}
+          >
+            <div>
+              <span
+                className={cx({
+                  [disabledDescriptionStyles]: disabled,
+                })}
+                style={{ gridArea: 'label' }}
+              >
+                {label}
+              </span>
+            </div>
+            {optional && (
```

Comment:
suggestion: in the leafygreen UI, the optional text is always right next in the field and not label. should we align that?

Reply:
@Sgrinfy what do you think? Should we make this a follow up too?

Reply:
I'm looking at the optional field for "Client Key Password", and the optional label is in the field like it should be.  @mabaasit what should we align?

Reply:
this whole thing came from the screenshot linked. the label on this input is getting cluttered with many options. 

For LG input the label is on the right and within the input field. I suggested to align these two.

But now I feel like it will not look good either on file upload button.

![image](https://user-images.githubusercontent.com/1305718/149943242-bdea4b1e-a1e2-4842-991a-f89388fb5eb5.png)

Reply:
ah yes, I understand now. I would love to align these. Is this select file component only used in the connect form?

![Screenshot 2022-01-18 at 14 41 27](https://user-images.githubusercontent.com/63339511/149948754-6f27d240-4299-42a4-b832-952b51fc0850.png)

Reply:
exactly. Its used in import to/export from collection modals as well.

Reply:
Is that recent screenshot what it should be updated to @Sgrinfy ?
I'd like to merge this and update the styles in a follow up I'm thinking

Reply:
@Anemy No, let's leave it like it is currently implemented. This select file component it's used in other places too in Compass and I don't want to have slightly different styles. We can revaluate in future how to improve it.